pkg/restore/restore.go

```diff
@@ -917,44 +930,60 @@ func (r *pvRestorer) executePVAction(obj *unstructured.Unstructured) (*unstructu
 		return obj, nil
 	}
 
-	// If we can't find a snapshot record for this particular PV, it most likely wasn't a PV that Ark
-	// could snapshot, so return early instead of trying to restore from a snapshot.
-	backupInfo, found := r.volumeBackups[pvName]
-	if !found {
-		return obj, nil
-	}
+	// TODO: Determine if we check for existence of old backup format here
 
-	// Past this point, we expect to be doing a restore
+	for _, snapshot := range r.volumeSnapshots {
```

Comment:
you should find the snapshot whose `PersistentVolumeName` matches `obj.GetName()` -- that's the one we should be restoring.